### PR TITLE
Makefile libsodium.mk

### PR DESCRIPTION
**Pull request by rezahousseini**
Adding the library libsodium (https://download.libsodium.org/doc/).

### CODE REVIEW

File: src/libsodium.mk [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+$(PKG)_SUBDIR   := libsodium-$($(PKG)_VERSION)
+$(PKG)_FILE     := libsodium-$($(PKG)_VERSION).tar.gz
+$(PKG)_URL      := https://download.libsodium.org/libsodium/releases/$($(PKG)_FILE)
+$(PKG)_URL_2    := https://github.com/jedisct1/libsodium/releases/download/$($(PKG)_VERSION)/$($(PKG)_FILE)
+$(PKG)_DEPS     := gcc
+
+define $(PKG)_BUILD
+    cd '$(1)' && CHOST='$(TARGET)' ./configure \
+        --prefix='$(PREFIX)/$(TARGET)'
+    $(MAKE) -C '$(1)' -j '$(JOBS)' install
```
Comment (starius) [inline src/libsodium.mk]: Please build with `-j '$(JOBS)'` and install with `-j '1'`.

File: src/libsodium.mk [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+$(PKG)_CHECKSUM := 940d03ea7d2caa7940e24564bf6d9f66d6edd1df1e0111ff8e3655f3b864fb59
+$(PKG)_SUBDIR   := libsodium-$($(PKG)_VERSION)
+$(PKG)_FILE     := libsodium-$($(PKG)_VERSION).tar.gz
+$(PKG)_URL      := https://download.libsodium.org/libsodium/releases/$($(PKG)_FILE)
+$(PKG)_URL_2    := https://github.com/jedisct1/libsodium/releases/download/$($(PKG)_VERSION)/$($(PKG)_FILE)
+$(PKG)_DEPS     := gcc
+
+define $(PKG)_BUILD
+    cd '$(1)' && CHOST='$(TARGET)' ./configure \
+        --prefix='$(PREFIX)/$(TARGET)'
```
Comment (starius) [inline src/libsodium.mk]: Please use `$(MXE_CONFIGURE_OPTS)` instead.

**Comment by starius** [comment]
Please squash two commits (bd0e50cdde72c392a4866193a63a62c36f65f721 and 799c37245f5924bff70a372d6e8f495de80344be). Adding a package is an atomic change.

**Comment by starius** [comment]
Please provide automatic updater of the package. See other `*.mk` files (section `$(PKG)_UPDATE`) to get examples.

File: src/libsodium.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    $(SED) -n 's,.*libsodium-\([0-9][^>]*\)\.tar.*,\1,p' | \
+    grep -v mingw | \
+    tail -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(1)' && CHOST='$(TARGET)' ./configure \
+        $(MXE_CONFIGURE_OPTS) \
+        --prefix='$(PREFIX)/$(TARGET)'
+    $(MAKE) -C '$(1)' -j '$(JOBS)' install -j '1'
```
Comment (tonytheodore) [inline src/libsodium.mk]: Put this on two lines, the `-j '1'` at the end overrides previous setting.

**Comment by tonytheodore** [comment]
Thanks!

Merged https://github.com/mxe/mxe/commit/ce796c1b3f439cd988b0ac8dfdba61457ca5a7b5 with some fixes for unnecessary options and separate install step https://github.com/mxe/mxe/commit/597fccc92103a17fec503db007bce0f61070a7bb.

**Pull request closed without merging**